android/tangram/jni/platform_android.cpp

```diff
@@ -311,6 +322,38 @@ void setCurrentThreadPriority(int priority) {
     setpriority(PRIO_PROCESS, tid, priority);
 }
 
+void labelsPickCallback(jobject listener, const std::vector<Tangram::TouchLabel>& labels) {
+
+    JniThreadBinding jniEnv(jvm);
+
+    float position[2] = {0.0, 0.0};
+
+    jobject touchLabel = nullptr;
+
+    if (labels.size() > 0) {
+        auto label = labels[0];
+        auto properties = label.touchItem.properties;
+
+        position[0] = label.touchItem.position[0];
+        position[1] = label.touchItem.position[1];
+
+        jobject hashmap = jniEnv->NewObject(hashmapClass, hashmapInitMID);
+
+        for (const auto& item : properties->items()) {
+            jstring jkey = jniEnv->NewStringUTF(item.key.c_str());
+            jstring jvalue = jniEnv->NewStringUTF(properties->asString(item.value).c_str());
+            jniEnv->CallObjectMethod(hashmap, hashmapPutMID, jkey, jvalue);
+        }
+
+        jdoubleArray darr = jniEnv->NewDoubleArray(2);
+        jniEnv->SetDoubleArrayRegion(darr, 0, 2, reinterpret_cast<jdouble*>(&label.coordinate));
```

Comment:
Probably easier to just use 2 double arguments here rather than instantiate an array.